autotest/gdrivers/plmosaic.py

```diff
@@ -179,6 +190,12 @@ def test_plmosaic_8():
 }""",
     )
 
+    yield
```

Comment:
Not necessarily to be done in this PR, but I'm thinking that it would be cleaner if those temp /vsimem/ files would be removed after the test is finished. So presumably as a cleanup step after the yield

An alternative would be to use
```
with gdaltest.tempfile("/vsimem/foo", "content"), gdaltest.tempfile("/vsimem/bar", "content"):
    yield
```
which takes care of the cleanup.

The same applies for the valid_mosaic fixture

Reply:
What do you think of the generic cleanup on line 56? That way, whatever mess a given test creates, subsequent tests will not be affected.

Reply:
> What do you think of the generic cleanup on line 56?

ah, I thought it was a module level setup_and_cleanup() like done usually, and didn't realize that it would apply to all tests. So ignore my above remark

Reply:
Maybe it should be renamed `setup_and_cleanup_test` ?

Somewhat related: if tests are running in parallel, they share a global `/vsimem` space, right? So to avoid potential collisions there might be use for some sort of `mem_temp_path` fixture?